script/e2e-test.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2018 Bitnami
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+ROOT_DIR=`cd "$( dirname "${BASH_SOURCE[0]}" )/.." >/dev/null && pwd`
+
+source $ROOT_DIR/script/libtest.sh
+
+# Add admin permissions to default user in kube-system namespace
+kubectl get clusterrolebinding kube-dns-admin >& /dev/null || \
+    kubectl create clusterrolebinding kube-dns-admin --serviceaccount=kube-system:default --clusterrole=cluster-admin 
+
+# Wait for Tiller
+k8s_wait_for_pod_ready -n kube-system -l app=helm,name=tiller
+wait_for_tiller
+
+# Install Kubeapps
+# TODO: Use just built images
+helm dep up $ROOT_DIR/chart/kubeapps/
+helm install --name kubeapps-ci --namespace kubeapps $ROOT_DIR/chart/kubeapps
+
+# Wait for Kubeapps Pods
+k8s_wait_for_pod_ready -n kubeapps -l app=kubeapps-ci
```

Comment:
maybe we can parameterize this a little bit?

For example extract the NAMESPACE the list of pods to wait and iterate?

Reply:
You can also list by namespace which is safer if we add a new tier. 

For example this code will wait for all deployments and replicas to rollout. And will retry a limited number of times.

```
         for i in {1..6}; do
           res=
             kubectl get deployment -o name \
               --context ${k8s_context} \
               --namespace kubeapps
             | xargs -n1 kubectl rollout status \
               --context ${k8s_context} \
               --namespace kubeapps
             || res=\$?
        
           if [ -z "\$res" ]; then
                exit
           fi
         done

```

Reply:
> You can also list by namespace which is safer if we add a new tier.

You mean list by `context`? (I guess so since I am already listing by namespace) We have that in other projects but I find it difficult to use locally since that parameter it's not usually set. I'd rather add it if at some point we need it.

> For example this code will wait for all deployments and replicas to rollout

I'd rather use pods here to support things like jobs or statefulsets. In any case it's true that's better to check the status per element.